src/youtube-player/youtube-player.ts

```diff
@@ -562,3 +586,37 @@ export class YouTubePlayer implements AfterViewInit, OnChanges, OnDestroy {
     );
   }
 }
+
+let apiLoaded = false;
+
+/** Loads the YouTube API from a specified URL only once. */
+function loadApi(): void {
+  if (apiLoaded) {
+    return;
+  }
+
+  // We can use `document` directly here, because this logic doesn't run outside the browser.
+  const url = 'https://www.youtube.com/iframe_api';
+  const script = document.createElement('script');
```

Comment:
The only problem is likely if a nonce would
be required, but in that case, the API can be loaded manually

Reply:
Ah good point. I've integrated it with the `CSP_NONCE` token.